src/etc/inc/plugins.inc.d/dhcpd.inc

```diff
@@ -268,44 +281,48 @@ function dhcpd_radvd_configure($verbose = false, $blacklist = array())
             }
         }
 
-        $radvdconf .= "# Generated for DHCPv6 server $dhcpv6if\n";
-        $radvdconf .= "interface {$realif} {\n";
-        $radvdconf .= "\tAdvSendAdvert on;\n";
-        $radvdconf .= sprintf("\tMinRtrAdvInterval %s;\n", !empty($dhcpv6ifconf['ramininterval']) ? $dhcpv6ifconf['ramininterval'] : '200');
-        $radvdconf .= sprintf("\tMaxRtrAdvInterval %s;\n", !empty($dhcpv6ifconf['ramaxinterval']) ? $dhcpv6ifconf['ramaxinterval'] : '600');
+        $rtadvdconf .= "# Generated for DHCPv6 server $dhcpv6if\n";
+
+        $rtadvd_interface_list[] = $realif;
+        $rtadvdconf .= "{$realif}:\x5c\n";
+        // $radvdconf .= "\tAdvSendAdvert on;\n"; // No equivemlent?  
+        $rtadvdconf .= sprintf("\t:mininterval#%s:\x5c\n", !empty($dhcpv6ifconf['ramininterval']) ? $dhcpv6ifconf['ramininterval'] : '200');
+        $rtadvdconf .= sprintf("\t:maxinterval#%s:\x5c\n", !empty($dhcpv6ifconf['ramaxinterval']) ? $dhcpv6ifconf['ramaxinterval'] : '200');
         if (!empty($dhcpv6ifconf['AdvDefaultLifetime'])) {
-            $radvdconf .= sprintf("\tAdvDefaultLifetime %s;\n", $dhcpv6ifconf['AdvDefaultLifetime']);
+            $rtadvdconf .= sprintf("\:rltime#%s:\x5c\n", $dhcpv6ifconf['AdvDefaultLifetime']);
         }
-        $radvdconf .= sprintf("\tAdvLinkMTU %s;\n", !empty($mtu) ? $mtu : 0);
+        $rtadvdconf .= sprintf("\t:mtu#%s:\x5c\n", !empty($mtu) ? $mtu : 0);
 
+        $rtaflags = "";
+        
         switch ($dhcpv6ifconf['rapriority']) {
             case "low":
-                $radvdconf .= "\tAdvDefaultPreference low;\n";
+                $rtaflags = 24;
```

Comment:
Is it possible to use hex for all of these flag bits?

Reply:
Makes no odds really, they are not exposed to the user.

Reply:
I know they are not user facing.
My comment is aimed at the maintainability for these flags being explicitly mentioned in hex in the freebsd / openbsd documentation. Makes skimming through this code easier when reading up on raflags.

But im not affiliated with opnsense, you might aswell wait what an official reviewer has to say here.

Reply:
Just my brain, I flip between hex and decimal and it depends on the mood I'm in. :)